mypy/newsemanal/semanal.py

```diff
@@ -1628,10 +1628,7 @@ def visit_import_from(self, imp: ImportFrom) -> None:
             # If it is still not resolved, check for a module level __getattr__
             if (module and not node and (module.is_stub or self.options.python_version >= (3, 7))
                     and '__getattr__' in module.names):
-                if self.type:
-                    fullname = self.type.fullname() + "." + imported_id
-                else:
-                    fullname = self.qualified_name(imported_id)
+                fullname = import_id + '.' + id
```

Comment:
This is a bit subtle, I would add a comment saying that we use the original full name, so that multiple imports refer to the same "original definition".